geohash.go

```diff
@@ -142,3 +152,41 @@ func (r Range) Round() float64 {
 	}
 	return roundDecimal(r.Mid(), dec)
 }
+
+// Return the neighbors of the supplied geohash. They are returned as a slice,
+// starting at the neighbor to the north, and going around clockwise with the
+// last being the neighbor to the northwest. See diagram below:
+//
+// 7  0  1
+// 6  x  2
+// 5  4  3
+
+func Neighbors(gh string) ([]string, error) {
+	neighbors := make([]string, 8)
+
+	box, err := Decode(gh)
+
+	if err != nil {
+		return nil, err
+	}
+	precision := len(gh)
+
+	// North
+	neighbors[0] = Encode(box.Lat.Mid() + box.Height(), box.Lon.Mid(), precision)
```

Comment:
`box.Lat.Mid() + box.Height()` => `box.Lat.Mid() + box.Lat.Val()`
Then you can remove `.Width()` and `.Height()`